pkg/devfile/adapters/docker/component/utils.go

```diff
@@ -187,6 +194,10 @@ func (a Adapter) startContainer(mounts []mount.Mount, projectVolumeName string,
 		Mounts: mounts,
 	}
 
+	hostConfig, err := a.generateAndGetHostConfig()
```

Comment:
You're overwriting the hostconfig added in the earlier step, which adds the volume mounts to the host config, causing the volumes defined in the devfile to not get mounted to the container. I think we need to rework this to either:
- Pass in the `mounts` object to `generateAndGetHostConfig`, or:
- Move the line that sets the mounts down to below this line.

Reply:
updated